nrf_security/src/drivers/zephyr/zephyr_entropy.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/entropy.h>
+#include <entropy_poll.h>
+
+#include "psa/crypto.h"
+
+psa_status_t zephyr_get_entropy(uint32_t flags,
+				size_t* estimate_bits,
+				uint8_t* output,
+				size_t output_size)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_entropy));
+	uint16_t request_len = MIN(UINT16_MAX, output_size);
+	int err;
+
+	/* Ignore flags */
+	(void)flags;
+
+	if (output == NULL || estimate_bits == NULL || output_size == 0) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (!device_is_ready(dev)) {
+		return PSA_ERROR_HARDWARE_FAILURE;
```

Comment:
We have an option to differentiate between driver not being ready and not getting entropy by using: 
PSA_ERROR_GENERIC_ERROR here.

The error-code is described as follows: "Implementations can use this error code if none of the other standard error codes are applicable."

From [here](https://armmbed.github.io/mbed-crypto/html/api/library/status.html?highlight=error%20codes#c.PSA_ERROR_GENERIC_ERROR)

Reply:
similar feature for mbedtls has the some error code `mbedtls_hardware_poll` in `entropy_poll.c`.
I don't see the error code being better, and if "device is not read" you essentially broke your hardware setup.
Should be easy to debug this.

As an improvement I think error logging is better. Can be added later.